csharp/ql/src/semmle/code/csharp/exprs/Call.qll

```diff
@@ -527,6 +527,39 @@ class MutatorOperatorCall extends OperatorCall {
   predicate isPostfix() { mutator_invocation_mode(this, 2) }
 }
 
+/**
+ * A function pointer or delegate call.
+ */
+abstract class DelegateLikeCall extends Call {
```

Comment:
We don't like `abstract` user-facing classes, where the user is not expected to contribute to the union. So instead you can do
```ql
private class DelegateLikeCall_ = @delegate_invocation_expr or @function_pointer_invocation_expr;

class DelegateLikeCall extends Call, DelegateLikeCall_ {
  ...
}
```